include/rdma/fi_log.h

```diff
@@ -35,6 +35,8 @@
 #ifndef _FI_LOG_H_
 #define _FI_LOG_H_
 
+#include "config.h"
```

Comment:
> The source code needs updating then.

I can add a commit to do that; I hadn't noticed the existing check in other files until you pointed it out.  IMO the right fix is simply to remove the `#if`/`#ifdef`, but it's mostly a style choice (we'll never *not* include `config.h` in practice).  You appear to care quite a bit more about this little bit of code than I do, so I can change it to be whichever way you would prefer.

> There's got to be a reason why this check is in all of the other OFA code.  Maybe custom build scripts?

Much older versions of autoconf (and apps using them) didn't rely so heavily on `config.h`.  This is a silly `#ifdef` for modern software that requires a version of autoconf >= 2.60 and unconditionally uses `AC_CONFIG_HEADERS`.  Also, there's a whole lot of copy/paste and cargo-culting that happens around the autotools for some reason (they're not really that complicated).